### PR TITLE
-Refactored the entire frontend codebase.

### LINKED ISSUE #6: Text Window for Dialogues (component)
**As a User**  
**I need** a clear and visually appealing text window for dialogues,  
**in order to** read and follow the story easily.

#### Description
Implement a `TextWindow.tsx` component that displays dialogues and textual information in a user-friendly and aesthetically pleasing manner.

#### Acceptance Criteria

- [x] Develop a `TextWindow.tsx` component that enhances readability and visual appeal.
- [x] Integrate the text window seamlessly into various scenes.

#### Tasks

- [x] Create a visually appealing design for the text window.
- [x] Ensure the text window is easily readable on different devices.
- [x] Test the text window in various scenes for consistency and visibility.
- [x] Implement styling that complements the overall design of the app.

### PR DESCRIPTION
**Pull request by BrainCodeJockey**
-Refactored the entire frontend codebase
-Performed CSS styling
-Updated the useEffect for state management
-Created a hook to ensure the location is always displayed

### CODE REVIEW

**Review by mpagels — CHANGES_REQUESTED** [review]
This pr is too big to review. 1000 changes? 37 files? Different topics that have changes.
Please split this pr in at least 4 different small pull request with changes that has one topic in common.

E.g.:
PR1 -> only changes that are relating to the refactoring
PR2 -> only changes that are relating to the css styling
PR3 -> only changes that are relating to the useEffect Update
PR4 -> only changes that are relating to the global state update

This way you make sure, that the reviewer can concentrate on only one thing and the cognitive overload is not that much. And the changes are higher, you get a review, if the changes are small. Like only 4 files and 50 lines of code to review instead of 37 files and 1000 changes...

**Comment by BrainCodeJockey** [comment]
> This pr is too big to review. 1000 changes? 37 files? Different topics that have changes. Please split this pr in at least 4 different small pull request with changes that has one topic in common.
> 
> E.g.: PR1 -> only changes that are relating to the refactoring PR2 -> only changes that are relating to the css styling PR3 -> only changes that are relating to the useEffect Update PR4 -> only changes that are relating to the global state update
> 
> This way you make sure, that the reviewer can concentrate on only one thing and the cognitive overload is not that much. And the changes are higher, you get a review, if the changes are small. Like only 4 files and 50 lines of code to review instead of 37 files and 1000 changes...

How do I split this large commit into smaller pushes?

**Comment by BrainCodeJockey** [comment]
The pull request has been split into three pieces.

File: frontend/src/components/DeleteSaveGame/DeleteSaveGame.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import React, { useState } from 'react';
 import axios from 'axios';
-import { SaveGameDto } from '../../types/types.ts';
+import { SaveGameDto } from '../../types/types';
 import './DeleteSaveGame.css';
 
 interface DeleteSaveGameProps {
     saveGame: SaveGameDto;
     onDeleteSaveGame: (saveId: string) => void;
+    onDeleteHero: (heroId: string) => void;
```
Comment (AljoschaNy) [inline frontend/src/components/DeleteSaveGame/DeleteSaveGame.tsx:9]: ist nur ein Hinweis: du hast bisher ausschließlich type genutzt. Deswegen würde ich dieses Interface noch zu einem type umwandeln und zu deiner types.ts hinzufügen

File: frontend/src/components/SceneLayout/SceneLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    path: string;
+    label: string;
+}
+
+interface SceneLayoutProps {
+    title: string;
+    navLinks: NavLink[];
+    children: ReactNode;
+    backgroundImage: string;
+}
```
Comment (AljoschaNy) [inline frontend/src/components/SceneLayout/SceneLayout.tsx:15]: Wie vorhin erwähnt, ich würde bei type bleiben und deine ganzen types in deiner zentralen types.ts verwalten

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,19 +1,72 @@
...
+
+        if (shouldUpdate) {
+            fetchDataAndUpdate()
+                .then(() => {
+                    setShouldUpdate(false);
+                })
+                .catch((error) => {
+                    console.error('Fehler beim Laden der Daten:', error);
+                });
+        }
```
Comment (AljoschaNy) [inline frontend/src/App.tsx]: Ich würde den Funktionsaufruf in einen Else Block einfügen. So gibt es im Fall, dass wenn shouldUpdate true ist, keine doppelten API-Aufrufe
```suggestion
        if (shouldUpdate) {
            fetchDataAndUpdate()
                .then(() => {
                    setShouldUpdate(false);
                })
                .catch((error) => {
                    console.error('Fehler beim Laden der Daten:', error);
                });
        } else {
         fetchDataAndUpdate()
            .catch((error) => {
                console.error('Fehler beim Laden der Daten:', error);
            });
        }
```

**Review by AljoschaNy — CHANGES_REQUESTED** [review]
Soweit sieht alles Top aus. Statt interface würde ich allerdings type nutzen.

**Review by AljoschaNy — APPROVED** [review]

**Comment by AljoschaNy** [comment]
LGTM, top!

**Pull request merged**